docs/src/pages/components/buttons/LoadingButtons.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      '& button': {
+        margin: theme.spacing(1),
+      },
+    },
+  }),
+);
+
+export default function LoadingButtons() {
+  const classes = useStyles();
+  const [loading, setLoading] = React.useState(false);
+
+  return (
+    <div className={classes.root}>
+      <Button variant="outlined" loading>
+        Loading…
+      </Button>
+      <Button
+        variant="contained"
+        color="primary"
+        loading
+        startIcon={<CircularProgress color="inherit" size={16} />}
```

Comment:
Most implementations bundle the circular directly in the button. I'm not confident this best, from a bundle size perspective. I'm wondering. It might also complexify the API to expose.

Reply:
This is a general statement not targeted at anyone particular: I'm getting  weary of this "props everything" approach. React introduced "component based ui development" not "props based ui development". It's ok to  add another `LoadableButton` where one could flip the switch and we provide the spinner.

Reply:
"BusyButton" 🙂